contract/src/main/java/io/neow3j/contract/NeoToken.java

```diff
@@ -197,30 +200,72 @@ public Map<ECPublicKey, BigInteger> getCandidates() throws IOException {
-            if (!nrItem.getType().equals(INTEGER)) {
-                throw new UnexpectedReturnTypeException(nrItem.getType(), INTEGER);
-            }
-            validators.put(key, nrItem.getInteger());
+            addMappingOfCandidateAndVote(validators, valItem);
         }
         return validators;
     }
 
     /**
      * Checks if there is a committee candidate or member with {@code publicKey}.
      *
-     * @param publicKey the candidates public key.
+     * @param publicKey the candidate's public key.
      * @return true if the public key belongs to a candidate. False otherwise.
      * @throws IOException if there was a problem fetching information from the Neo node.
      */
     public boolean isCandidate(ECPublicKey publicKey) throws IOException {
         return getCandidates().containsKey(publicKey);
     }
 
+    /**
+     * Gets the public keys of all currently registered candidates and their corresponding vote count.
+     * <p>
+     * The vote count is based on the summed up NEO balances of the respective candidate's voters.
+     * <p>
+     * The response contains an iterator which might be truncated based on the size configuration of the node this
+     * request is sent to.
+     *
+     * @return the candidate public keys and their corresponding vote count.
+     * @throws IOException                   if there was a problem fetching information from the Neo node.
+     * @throws UnexpectedReturnTypeException if the return type is not an interop interface or the iterator elements
+     *                                       are not public keys and node counts.
+     */
+    public Map<ECPublicKey, BigInteger> getAllCandidates() throws IOException {
+        StackItem stackItem = callInvokeFunction(GET_ALL_CANDIDATES).getInvocationResult().getStack().get(0);
+        if (!stackItem.getType().equals(INTEROP_INTERFACE)) {
+            throw new UnexpectedReturnTypeException(stackItem.getType(), INTEROP_INTERFACE);
+        }
+        Map<ECPublicKey, BigInteger> candidates = new HashMap<>();
+        for (StackItem iteratorItem : stackItem.getIterator()) {
+            addMappingOfCandidateAndVote(candidates, iteratorItem);
+        }
+        return candidates;
+    }
+
+    // Extracts the candidates public key and its votes from the stack item, and adds a corresponding mapping to the
+    // provided Map instance.
+    private void addMappingOfCandidateAndVote(Map<ECPublicKey, BigInteger> candidates, StackItem candItem) {
+        if (!candItem.getType().equals(STRUCT)) {
+            throw new UnexpectedReturnTypeException(candItem.getType(), STRUCT);
+        }
+        ECPublicKey key = extractPublicKey(candItem.getList().get(0));
+        StackItem nrItem = candItem.getList().get(1);
+        if (!nrItem.getType().equals(INTEGER)) {
+            throw new UnexpectedReturnTypeException(nrItem.getType(), INTEGER);
+        }
+        candidates.put(key, nrItem.getInteger());
+    }
```

Comment:
Minor detail: The candidates map should be created inside of this method instead of taking it as a parameter and modifying it. It's not bad in this case but usually I wouldn't want a method to have suche "side effects", i.e., taking a value as parameter and modifying it without the caller expecting it.

Reply:
Noted, I'll improve that part.